application-faq-ui/src/main/resources/FAQCode/FAQHomeSheet.xml

```diff
@@ -98,7 +98,7 @@
       {{html clean='false'}}
         $services.localization.render('platform.faq.need', ["&lt;a href=$login&gt;","&lt;/a&gt;","&lt;a href=$register&gt;","&lt;/a&gt;"])
       {{/html}}
-    #else
+    #elseif ($services.security.authorization.hasAccess("edit", "${faqSpace}.WebHome"))
```

Comment:
Small detail: always use the single quote in velocity when there's no evaluation to be done inside the string. This makes velocity perform better since it can skip the eval on the content: `"edit"` --> `'edit'`.

Side note: We should not use the magical `WebHome` anymore, nor manipulate directly reference as strings. It's better to go through the Reference resolvers (`$services.render....`).

I'm going to apply it but it would be nice to fix these 2 points.

Thanks Stephane for the PR!

Reply:
BTW are nested spaces supported?

Reply:
Vincent, I used single quotes instead in the new commit. I usually try to double quote strings only when needed indeed, this time I got misdrawn by the [Security Module documentation](https://extensions.xwiki.org/xwiki/bin/view/Extension/Security%20Module), which I hence just updated as well. 

Regarding the "WebHome" reference: I understand the issue, however it's used in another place by the application at the moment (in the "parent" input form), so I would rather propose we create a new issue specifically about it and handle references and possibly nested spaces it that scope, what do you think?

Reply:
> Regarding the "WebHome" reference: I understand the issue, however it's used in another place by the application at the moment (in the "parent" input form), so I would rather propose we create a new issue specifically about it and handle references and possibly nested spaces it that scope, what do you think?

We were definitely not following this practice before (we didn't even have the tools in the past ;)) but our general strategy is:
* Do not increase the technical debt by using bad practices in new code
* Try, when possible and when time permits, refactor old code to use the new best practices

Consequence: we should absolutely never wait for all the places to be fixed at the same to use the new best practices or it will never happen and we'll 1) keep increasing the technical debt and 2) make it even harder to use the new best practices since more bad code will have been added :)

Sure you can create a new issue (and provide a PR there if you want if you have the time) ;)

Thanks again

Reply:
Vincent, Right. I updated the code so as to not increase the technical debt indeed. I could not figure out how to serialize the home page of a space without using the "WebHome" supposition though, in order to replace the string `parent="${faqSpace}.WebHome"` by one that is generated by the model module. What would be the right way to do so? (the [Model Module documentation](https://extensions.xwiki.org/xwiki/bin/view/Extension/Model%20Module) contains "WebHome" several times, should it be differently?)

Reply:
I was fearing that you were going to ask ;)

Ok I'll spend the time to provide the PR then!

Reply:
:-)

Reply:
So first the API is: `public boolean hasAccess(Right right, EntityReference entityReference)` so you can pass any EntityReference.

Second, see the last example on https://extensions.xwiki.org/xwiki/bin/view/Extension/Model%20Module#HScriptServiceAPI which I've just added, with explanations:

```
$services.model.resolveDocument('', 'default', $doc.documentReference.lastSpaceReference)
```

Thanks!

Reply:
Side note: This is not very obvious and we could imagine adding some helper method to ModelScriptService to make this use case simpler for dev users.

Reply:
Thank you Vincent, I used your hint in the latest commit 985b38e88ee15d8c2ff152be24fa38fce970e2dd, hoping it's ok, I'll see if a release can be performed.